draft-gpew-priv-ppm.md

```diff
@@ -672,242 +679,366 @@ independently.
 ~~~
 
-Next, the helper encrypts the aggregate share `agg_share` under the collector's
-public key as follows:
+where `inbound` is the previous VDAF message sent by the leader and `prep_state` is
+its current preparation state. If this operation fails, then the helper fails
+with error `vdaf-prep-error`. Otherwise, it interprets `out` as follows. If this
+is the last round of VDAF preparation phase, then `out` is the helper's output
+share, in which case the helper finishes and transitions to FINISHED. Otherwise,
+the helper interpets `out` as the tuple `(new_state, agg_msg)`, where
+`new_state` is its updated preperation state and `agg_msg` is its next VDAF
+message.
+
+This output message for each report in AggregateContinueReq.seq is then sent to the leader
+in an AggregateContinueResp message, structured as follows:
 
 ~~~
-enc, context = SetupBaseS(pk, task_id,
-                          "ppm-00 aggregate share" || 0x03 || 0x00)
-encrypted_agg_share = context.Seal(batch_interval, agg_share)
+struct {
+  opaque helper_state<0..2^16>;
+  ProcessShare seq<1..2^16-1>;
+} AggregateContinueResp;
 ~~~
 
-where `pk` is the HPKE public key encoded by the collector's HPKE key
-configuration and `task_id` is `AggregateShareReq.task_id`. `agg_share` is the
-serialized `AggregateShare`, and `batch_interval` is obtained from the
-`AggregateShareReq`.
+The order of AggregateContinueResp.seq matches that of the InputReportShare values in
+`AggregateContinueReq.seq`. The helper's response to the leader is an HTTP 200 OK whose body
+is the AggregateInitResp and media type is "message/ppm-aggregate-continue-resp". The helper
+then awaits the next message from the leader.
+
+### Aggregate Completion {#agg-complete}
 
-This encryption prevents the leader from learning the actual result, as it only
-has its own share and not the helper's share, which is encrypted for the
-collector. The helper responds to the leader with HTTP status 200 OK and a body
-consisting of an `HpkeCiphertext` where `config_id` is set to the collector's
-HPKE config ID, `enc` is set to the encapsulated HPKE context `enc` and
-`ciphertext` is `encrypted_agg_share`.
+Once processing of a report share is finished, each aggregator stores the the
+recovered output shares until the batch to which it pertains is collected.
+To aggregate the output shares, denoted `out_shares`, the aggregator runs
+the aggregation algorithm specified by the VDAF:
 
-The leader uses the helper's aggregate share response to respond to the
-collector's collect request (see {{pa-collect}}).
+~~~
+agg_share = VDAF.out_shares_to_agg_share(agg_param, out_shares)
+~~~
 
-## Collecting Results {#pa-collect}
+Note that for most VDAFs, it is possible to aggregate output shares as they
+arrive rather than wait until the batch is collected. To do so however, it is
+necessary to enforce the batch parameters as described in
+{{batch-parameter-validation}} so that the aggregator knows which aggregate
+share to update.
```

Comment:
This is kind of awkward. The implication is that batch parameters don't need to be validated when using a VDAF without an aggregation parameter. Also I don't understand how validation of batch parameters informs which aggregate is used?